src/lib/annotations/doc/CreateHighlightDialog.js

```diff
@@ -323,42 +353,52 @@ class CreateHighlightDialog extends EventEmitter {
         if (this.isMobile) {
             highlightDialogEl.classList.add('bp-mobile-annotation-dialog');
             highlightDialogEl.classList.add('bp-annotation-dialog');
-            highlightDialogEl.querySelector('.bp-annotation-caret').remove();
         }
 
         const containerEl = highlightDialogEl.querySelector(constants.SELECTOR_ANNOTATION_HIGHLIGHT_DIALOG);
 
         // Reference HTML
-        this.highlightCreateEl = containerEl.querySelector(constants.SELECTOR_ADD_HIGHLIGHT_BTN);
-        this.commentCreateEl = containerEl.querySelector(`.${constants.CLASS_ADD_HIGHLIGHT_COMMENT_BTN}`);
         this.buttonsEl = containerEl.querySelector(constants.SELECTOR_HIGHLIGHT_BTNS);
 
-        // Create comment box
-        this.commentBox = new CommentBox(containerEl);
-
         // Stop interacting with this element from triggering outside actions
         highlightDialogEl.addEventListener('click', this.stopPropagation);
         highlightDialogEl.addEventListener('mouseup', this.stopPropagation);
         highlightDialogEl.addEventListener('dblclick', this.stopPropagation);
 
-        // Event listeners
-        this.highlightCreateEl.addEventListener('click', this.onHighlightClick);
-        this.commentCreateEl.addEventListener('click', this.onCommentClick);
-        this.commentBox.addListener(CommentBox.CommentEvents.post, this.onCommentPost);
-        this.commentBox.addListener(CommentBox.CommentEvents.cancel, this.onCommentCancel);
+        // Events for highlight button
+        if (this.allowHighlight) {
+            this.highlightCreateEl = containerEl.querySelector(constants.SELECTOR_ADD_HIGHLIGHT_BTN);
+            this.highlightCreateEl.addEventListener('click', this.onHighlightClick);
+
+            if (this.hasTouch) {
+                this.highlightCreateEl.addEventListener('touchstart', this.stopPropagation);
```

Comment:
maybe move the adding of these 2 event handlers into its own method? so you can just call it anywhere on w/e element you pass in?

Reply:
I'm not sure this'd be valid for the `touchend` -> `highlightClick` event binding, as it only ever happens once. Would it be valid abstracting the binding for `stopPropagation()`?

Maybe instead of 
```
element.addEventListener('event', this.stopPropagation);
```
We'd do
```
this.bindStopPropagation(element, event);
```

Though, after seeing it, I'm not sure what the benefit of that is

Reply:
hmm true. I guess it wouldn't really clean up too much.